libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -180,19 +180,72 @@ GCS_MAVLINK::setup_uart(const AP_SerialManager& serial_manager, AP_SerialManager
  * @brief Send the next pending waypoint, called from deferred message
  * handling code
  */
-void
-GCS_MAVLINK::queued_mission_request_send()
+void MissionItemProtocol::queued_request_send()
 {
-    if (initialised &&
-        waypoint_receiving &&
-        waypoint_request_i <= waypoint_request_last) {
-        mavlink_msg_mission_request_send(
-            chan,
-            waypoint_dest_sysid,
-            waypoint_dest_compid,
-            waypoint_request_i,
-            MAV_MISSION_TYPE_MISSION);
+    if (link == nullptr) {
+        AP::internalerror().error(AP_InternalError::error_t::gcs_bad_missionprotocol_link);
+        return;
+    }
+    if (!receiving) {
+        return;
+    }
+    if (request_i > request_last) {
+        return;
+    }
+    mavlink_msg_mission_request_send(
+        link->get_chan(),
+        dest_sysid,
+        dest_compid,
+        request_i,
+        mission_type());
+    timelast_request_ms = AP_HAL::millis();
+}
+
+void MissionItemProtocol::update()
+{
+    if (!receiving) {
+        // we don't need to do anything unless we're sending requests
+        return;
+    }
+    if (link == nullptr) {
+        AP::internalerror().error(AP_InternalError::error_t::gcs_bad_missionprotocol_link);
+        return;
     }
+    const uint32_t wp_recv_time = 1000U + (link->get_stream_slowdown_ms()*20);
```

Comment:
no need to add a _ms or _us on the end of this variable name?

Reply:
Yes.  Actually, I'll just kill the temporary variable.